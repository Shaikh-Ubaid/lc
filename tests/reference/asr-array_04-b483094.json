{
    "basename": "asr-array_04-b483094",
    "cmd": "lc --show-asr --no-color {infile} -o {outfile} -extra-arg=\"\"",
    "infile": "tests/../integration_tests/array_04.cpp",
    "infile_hash": "dd38786061d4a19743e4c32c2a3ecb08193777f81a2a0d3e87af0f0a",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_04-b483094.stdout",
    "stdout_hash": "527842cbdfdfac8d3cf494d61c760428aa51ec431fcf2b3e5d2827cd",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}