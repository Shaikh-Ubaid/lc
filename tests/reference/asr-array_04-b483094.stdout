(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Function
                    (SymbolTable
                        2
                        {
                            __return_var:
                                (Variable
                                    2
                                    __return_var
                                    []
                                    ReturnVar
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr1:
                                (Variable
                                    2
                                    arr1
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())
                                            (()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr2:
                                (Variable
                                    2
                                    arr2
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            res:
                                (Variable
                                    2
                                    res
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    (FunctionType
                        []
                        (Integer 4)
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(Allocate
                        [((Var 2 arr1)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))
                        ((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (=
                        (Var 2 arr1)
                        (ArrayReshape
                            (ArrayConstant
                                [(RealConstant
                                    1.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    3.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    5.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    7.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    5.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    7.000000
                                    (Real 8)
                                )]
                                (Array
                                    (Real 8)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 9 (Integer 4)))]
                                    FixedSizeArray
                                )
                                RowMajor
                            )
                            (ArrayPhysicalCast
                                (ArrayConstant
                                    [(ArraySize
                                        (Var 2 arr1)
                                        (IntegerConstant 1 (Integer 4))
                                        (Integer 4)
                                        ()
                                    )
                                    (ArraySize
                                        (Var 2 arr1)
                                        (IntegerConstant 2 (Integer 4))
                                        (Integer 4)
                                        ()
                                    )]
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 0 (Integer 4))
                                        (IntegerConstant 2 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    RowMajor
                                )
                                FixedSizeArray
                                DescriptorArray
                                (Array
                                    (Integer 4)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 2 (Integer 4)))]
                                    DescriptorArray
                                )
                                ()
                            )
                            (Array
                                (Real 8)
                                [(()
                                ())
                                (()
                                ())]
                                FixedSizeArray
                            )
                            ()
                        )
                        ()
                    )
                    (Allocate
                        [((Var 2 arr2)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (=
                        (Var 2 arr2)
                        (ArrayConstant
                            [(RealConstant
                                5.000000
                                (Real 8)
                            )
                            (RealConstant
                                6.000000
                                (Real 8)
                            )
                            (RealConstant
                                7.000000
                                (Real 8)
                            )]
                            (Array
                                (Real 8)
                                [((IntegerConstant 0 (Integer 4))
                                (IntegerConstant 3 (Integer 4)))]
                                FixedSizeArray
                            )
                            RowMajor
                        )
                        ()
                    )
                    (Allocate
                        [((Var 2 res)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (=
                        (Var 2 res)
                        (RealBinOp
                            (ArraySection
                                (Var 2 arr1)
                                [(()
                                (IntegerConstant 1 (Integer 4))
                                ())
                                ((ArrayBound
                                    (Var 2 arr1)
                                    (IntegerConstant 2 (Integer 4))
                                    (Integer 4)
                                    LBound
                                    ()
                                )
                                (ArrayBound
                                    (Var 2 arr1)
                                    (IntegerConstant 2 (Integer 4))
                                    (Integer 4)
                                    UBound
                                    ()
                                )
                                (IntegerConstant 1 (Integer 4)))]
                                (Array
                                    (Real 8)
                                    [(()
                                    ())]
                                    DescriptorArray
                                )
                                ()
                            )
                            Add
                            (Var 2 arr2)
                            (Array
                                (Real 8)
                                [(()
                                ())]
                                DescriptorArray
                            )
                            ()
                        )
                        ()
                    )
                    (Print
                        [(Var 2 arr1)
                        (Var 2 arr2)]
                        ()
                        ()
                    )
                    (Print
                        [(Var 2 res)]
                        ()
                        ()
                    )
                    (=
                        (Var 2 __return_var)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (Return)]
                    (Var 2 __return_var)
                    Public
                    .false.
                    .false.
                    ()
                ),
            main_program:
                (Program
                    (SymbolTable
                        3
                        {
                            exit_code:
                                (Variable
                                    3
                                    exit_code
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main_program
                    [main]
                    [(=
                        (Var 3 exit_code)
                        (FunctionCall
                            1 main
                            1 main
                            []
                            (Integer 4)
                            ()
                            ()
                        )
                        ()
                    )]
                )
        })
    []
)
